pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -159,16 +159,21 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{Requeue: false}, nil
 	}
 
+	patch := client.MergeFrom(deployment.DeepCopy())
 	d := &PmemCSIDriver{deployment, r.namespace, r.k8sVersion}
 
-	// update status
+	// update status and deployment
 	defer func() {
 		klog.Infof("Updating deployment status....")
 		d.Deployment.Status.LastUpdated = metav1.Now()
 		if statusErr := r.client.Status().Update(context.TODO(), d.Deployment); statusErr != nil {
 			klog.Warningf("failed to update status %q for deployment %q: %v",
 				d.Deployment.Status.Phase, d.Name, statusErr)
 		}
+
+		if err := r.client.Patch(context.TODO(), d.Deployment, patch); err != nil {
+			klog.Warningf("Failed update deployment %q: %s", d.GetName(), err)
+		}
```

Comment:
This doesn't belong into this commit, does it?

Reply:
It is to fix the issue observed in test failure in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-718/12/tests

```
Sep 9 13:36:13.470: INFO: got expected driver deployment pmem-csi-with-defaults
pmem-csi-operator-84b674657d-4xsj8/pmem-csi-operator@pmem..ker2: W0909 13:36:13.457821 1 deployment_controller.go:172] failed to update status "Running" for deployment "pmem-csi-with-defaults": Operation cannot be fulfilled on deployments.pmem-csi.intel.com "pmem-csi-with-defaults": the object has been modified; please apply your changes to the latest version and try again
```

Reply:
So the change is needed for the PR, but it still doesn't belong into this commit, right?

Should this hunk here be in the "patch deployment status instead of update"?

Or if it is unrelated to that one, put it into a separate commit with a separate explanation. As it stands now, nothing in the "delete obsolete objects of a deployment" commit message explains why this code here is being changed.

Sorry for being pedantic, but after having let an unrelated commit through in another PR I am now taking a stricter position.

Reply:
Sorry, this change is not related to the above-mentioned issue of updating status. It was part initial plan of updating deployment with annotations. I reverted this.